### ISSUE TITLE
Centered tables from code using typst

### ISSUE THREAD

**Issue by mcanouil** [quarto-dev/quarto-cli#8797]
From a simpler example:

Note that you can use `keep-typ` to investigate the Typst code.

Let's take a simpler/smaller example.

````qmd
---
format: typst
keep-typ: true
engine: knitr
---

| Name | Age |
|------|-----|
| John | 23  |


```{r}
knitr::kable(data.frame(name = "John", age = 23))
```
`````

This will produce the following code for both tables:

<table>
<tr><th>Markdown table</th><th>Generated table</th></tr>
<tr><td>

````typst
#figure(
align(center)[#table(
  columns: 2,
  align: (col, row) => (auto,auto,).at(col),
  inset: 6pt,
  [Name], [Age],
  [John],
  [23],
)]
)
````

</td><td>

````typst
#block[
```r
knitr::kable(data.frame(name = "John", age = 23))
```

#block[ #<- This is the issue causing the table to be not centered
#figure(
align(center)[#table(
  columns: 2,
  align: (col, row) => (left,right,).at(col),
  inset: 6pt,
  [name], [age],
  [John],
  [23],
)]
)

]
]
````

</td></tr>
</table>

### Discussed in https://github.com/quarto-dev/quarto-cli/discussions/8768

<div type='discussions-op-text'>

<sup>Originally posted by **psads-git** February 17, 2024</sup>
Consider the following `quarto` document. If `format` is `pdf`, the output is a centered table. However, when `format` is `typst`, the resulting table is left-aligned. Is it possible to get the table centered with `format: typst` likewise as with `format: pdf`? Thanks!

````
---
format: typst
papersize: a4
fontsize: 12pt
margin:
  x: 2.54cm
  y: 2.54cm
jupyter: python3
---

```{python}
import seaborn as sns

iris = sns.load_dataset('iris')
iris.head()
```
````</div>

**Comment by mcanouil** [comment]
This is slightly related to #8767

**Comment by gordonwoodhull** [comment]
Hmm, I had a different reason to replace the block with theorem instead of putting the theorem inside the block, I can try that here.

**Comment by gordonwoodhull** [comment]
I haven't dug too deep into the code yet, but I did some testing and reading of docs.

I haven't found any way to set the alignment of a table & wanted to confirm if this is expected. It seems to be either centered in PDF or 100% in HTML. (Except in the case of this bug, where it is left.)

I tried `fig-align` because it seems that a table is sometimes a figure (?), but it gets ignored and passes to the output.

````

| Default | Left | Right | Center |
|---------|:-----|------:|:------:|
| 12      | 12   |    12 |   12   |
| 123     | 123  |   123 |  123   |
| 1       | 1    |     1 |   1    |

: a table {fig-align="right"}
````

<img width="304" alt="image" src="https://github.com/quarto-dev/quarto-cli/assets/1366709/3e4045b3-1863-4a85-8328-e0de5d980125">

**Comment by cscheid** [comment]
> I tried fig-align because it seems that a table is sometimes a figure (?)

A table can be part of a _float_, and the content of a Figure float can be a table, but if we do "automatic discovery" like above (where you don't explicitly mark the type of float), then the float should be a Table, and it will contain a table. In that case, the alignment directive should be `tbl-align` instead of `fig-align`

(`tbl-`, `fig-`, `lst-` all refer to the type of float, respectively Table, Figure, and Listing. By default, Tables will contain tables, Figures will contain images, and Listings will contain code listings. But we need to allow images inside a Table float, because that might be the best way to express the Table they seek to see in the output.)

**Comment by gordonwoodhull** [comment]
Cool, I'll try that and build out FF matrix rows/cases. 

Got it, tables and figures are *floats*, they're just handled by the *figure* function in Typst. 

It's neat how the directive prefixes mirror the crossref prefixes.

**Comment by cscheid** [comment]
> Got it, tables and figures are floats, they're just handled by the _figure_ function in Typst.

Ah yes, sorry! In Typst, what we call "floats" are "figures", to make everything yet more confusing. (and we can't control that because it's their entry point to their crossref system)

**Comment by gordonwoodhull** [comment]
> the alignment directive should be `tbl-align` instead of `fig-align`

**\*would** be, it seems

To resolve this issue, I can either 
1. make tables centered in all cases for Typst
2. implement `tbl-align` and make Typst the only conforming format for the feature

I somewhat prefer (2) but have to check the effort/risk.

**Comment by gordonwoodhull** [comment]
Not at all surprised that there are format- (and engine- ?) specific workarounds for table alignment, see #8993

**Comment by cderv** [comment]
@gordonwoodhull I saw your comment on #9602, possibly fixing  this. 

Trying to add a test for it, I am finding that 

* Initial report is fixed in pre-release already (possibly by Pandoc updates)

	<table>
	<tr>
	 <td>Qmd
	 <td>PDF
	<tr>
	 <td>
	
	````markdown
	---
	format: typst
	keep-typ: true
	engine: knitr
	---
	
	| Name | Age |
	|------|-----|
	| John | 23  |
	
	
	```{r}
	knitr::kable(data.frame(name = "John", age = 23))
	```
	````
	
	
	 <td>
	
	![image](https://github.com/quarto-dev/quarto-cli/assets/6791940/934ad2f5-c2f2-4d6c-8887-ac24cf149e24)
	
	</table>

	So it seems tables will now be centered by default in Typst with Quarto 1.5.

* This leaves us with the `fig-align` / `tbl-align` option topic

So maybe we can close and open a clean issue with what is left, based on the feature matrix status for float alignement. I don't think alignment of table is a thing right now.

**Comment by gordonwoodhull** [comment]
Thanks @cderv. You are correct, the original test case was fixed by ab6a94d2.

#9602 fixes cases like 

````
```{python}
from great_tables import GT, md, html
from great_tables.data import islands

islands_mini = islands.head(10)

# Create a display table showing ten of the largest islands in the world
gt_tbl = GT(islands_mini)

# Show the output table
gt_tbl
```
````
